### PR TITLE
Update build scripts

### LINKED ISSUE #588: Integrate React Hot Loader v3
https://github.com/gaearon/redux-devtools/commit/64f58b7010a1b2a71ad16716eb37ac1031f93915

### PR DESCRIPTION
**Pull request by frenzzy**
**Do not merge!**
Work in progress..
- Do not bundle server-side code by [webpack](https://github.com/webpack/webpack), only [transpile](https://en.wikipedia.org/wiki/Source-to-source_compiler) via [babel](https://babeljs.io/) (this is the main goal ⚽ )
- Parallel build tasks execution, improved build performance
- Replace [React Transform](https://github.com/gaearon/babel-plugin-react-transform) with [React Hot Loader](https://github.com/gaearon/react-hot-loader) v3, closes #588
- New task: deploy static version to [GitHub Pages](https://pages.github.com/)
- Use original [webpack-dev-middleware](https://github.com/webpack/webpack-dev-middleware) instead of patched [webpack-middleware](https://github.com/kriasoft/webpack-middleware)
- Reuse the result of file and style loaders in the server-side code

TODO:
- Add watch mode to the `copy` and `compileServer` tasks

### CODE REVIEW

**Comment by langpavel** [comment]
Can you give a summary?

**Comment by jifeon** [comment]
> The build automation scripts rewritten without the babel-node usage in ES2015 flavor and require Node.js v6 (or newer) to run

What's the profit? I think it's cool when all JS files in a project use the same lang version. `async/await` in tools looks more readable than the `.then` chains.

**Comment by frenzzy** [comment]
@jifeon yes, you are right! I will roll it back. Also babel-node approach will allow to save backward compatibility with node v4 and v5 with the corresponding [babel plugins](https://babeljs.io/docs/plugins/).

Initially I just thought about consistency with other starter-kits like:
- [react-static-boilerplate / tools](https://github.com/koistya/react-static-boilerplate/tree/master/tools)
- [aspnet-starter-kit / tools](https://github.com/kriasoft/aspnet-starter-kit/tree/master/tools)
- [fsharp-starter-kit / tools](https://github.com/kriasoft/fsharp-starter-kit/tree/master/tools)

File: tools/copy.js [hunk 1]
```diff
@@ -1,51 +1,44 @@
...
  * This source code is licensed under the MIT license found in the
  * LICENSE.txt file in the root directory of this source tree.
  */
 
-import path from 'path';
-import gaze from 'gaze';
-import Promise from 'bluebird';
-import fs from './lib/fs';
-import pkg from '../package.json';
+const task = require('./task');
```
Comment (koistya) [inline tools/copy.js]: References to custom modules should go below the Node.js and 3rd party modules.

**Comment by codetony25** [comment]
Any updates on this this pull request?

File: tools/compileServer.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import path from 'path';
+const babel = require('babel-core');
+import { readFile, writeFile, copyFile, readDir, makeDir } from './lib/fs';
+
+/**
+ * Compile server-side application from the source files.
+ */
+async function compileServer() {
+  const dirs = await readDir('**/*.{js,json,md}', {
+    cwd: 'src',
```
Comment (codetony25) [inline tools/compileServer.js]: Should we be concerned with .jsx extensions too?

**Comment by ulani** [comment]
@frenzzy  thank you very much for this PR! Unfortunately, we have close it due to inactivity. Feel free to re-open it or [join](https://discord.com/invite/2nKEnKq) our [Discord](https://discord.com/invite/2nKEnKq) channel for [discussion](https://github.com/kriasoft/react-starter-kit/discussions/1950).

NOTE: The `main` branch has been updated with React Starter Kit v2, using JAM-style architecture.

**Pull request closed without merging**